style-guide.md

```diff
@@ -620,3 +620,45 @@ things = create_things()
 
 # res = triple_len(things)
 ```
+
+# Docstrings
+
+## Preferred delimiters
+
+You **should** enclose docstrings with `"""` delimiters.  That is, you **should
+not** enclose docstrings with `'''` delimiters.
+
+## Layout
+
+You **should** lay out docstring content according to the conventions defined
+in [PEP 257](https://www.python.org/dev/peps/pep-0257/) and [PEP
+287](https://www.python.org/dev/peps/pep-0287/).  Where those two PEPs
+conflict, precedence is given to the recommendations of [PEP
+287](https://www.python.org/dev/peps/pep-0287/).
+
+## Exceptions to docstring style PEPs
+
+The aforementioned [PEP 257](https://www.python.org/dev/peps/pep-0257/)
+recommends that a summary of a method's functionality should appear on the
+first line of a docstring immediately following the opening `"""` delimiter.
+If a team member so wishes, this summary **may** appear on the line immediately
+following the opening `"""` delimiter.  For example,
+
+```python
+# This is fine
+def foo():
+    """Summary of foo's functionality.
```

Comment:
Is this style of starting immediately on the same line as the `"""` used much in any of our libraries? I haven't seen that being used **except** for when the entire docstring fits into a single line (I personally dislike it even there haha).

If that style isn't used anywhere in our libraries already, I would rather not promote it here.

No strong feeling, just my personal biased opinion.

Reply:
Same.  I'm generally :-1: on this style though I do recognize it's common.  I find it harder to read than keeping the `"""` on their own lines.

Reply:
Okay sure let's make that restriction. I originally said it should be okay because I've seen it a lot in other codebases, it's defined in a PEP, and I didn't care that much if it popped up here and there but I do prefer the other style myself.